docs/writing-tests/test-runner.md

```diff
@@ -0,0 +1,293 @@
+| `--no-stories-json`             | Disables stories json mode <br/>`test-storybook --no-stories-json`                                                               |
+| `-c`, `--config-dir [dir-name]` | Directory where to load Storybook configurations from <br/>`test-storybook -c .storybook`                                        |
+| `--watch`                       | Run in watch mode <br/>`test-storybook --watch`                                                                                  |
+| `--url`                         | Define the URL to run tests in. Useful for custom Storybook URLs <br/>`test-storybook --url http://the-storybook-url-here.com`   |
+| `--browsers`                    | Define browsers to run tests in. One or multiple of: chromium, firefox, webkit <br/>`test-storybook --browsers firefox chromium` |
+| `--maxWorkers [amount]`         | Specifies the maximum number of workers the worker-pool will spawn for running tests <br/>`test-storybook --maxWorkers=2`        |
+| `--no-cache`                    | Disable the cache <br/>`test-storybook --no-cache`                                                                               |
+| `--clearCache`                  | Deletes the Jest cache directory and then exits without running tests <br/>`test-storybook --clearCache`                         |
+| `--verbose`                     | Display individual test results with the test suite hierarchy <br/>`test-storybook --verbose`                                    |
+| `-u`, `--updateSnapshot`        | Use this flag to re-record every snapshot that fails during this test run <br/>`test-storybook -u`                               |
+| `--eject`                       | Creates a local configuration file to override defaults of the test-runner <br/>`test-storybook --eject`                         |
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-test-runner-execute-with-flags.yarn.js.mdx',
+    'common/storybook-test-runner-execute-with-flags.npm.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+### Run tests against a deployed Storybook
+
+By default, the test runner assumes that you're running it against a locally served Storybook on port `6006`. If you want to define a target URL to run against deployed Storybooks, you can use the `--url` flag or set the `TARGET_URL` environment variable. For example:
+
+<!-- prettier-ignore-start -->
+
+<CodeSnippets
+  paths={[
+    'common/storybook-test-runner-execute-with-url.yarn.js.mdx',
+    'common/storybook-test-runner-execute-with-url.npm.js.mdx',
+    'common/storybook-test-runner-execute-with-url.env-var.js.mdx',
+  ]}
+/>
+
+<!-- prettier-ignore-end -->
+
+## Set up CI to run tests
+
+You can also configure the test runner to run tests on a CI environment. Documented below are some recipes to help you get started.
+
+### Run against deployed Storybooks
```

Comment:
This (either this heading or the following paragraph) is a little misleading, I think. If the Storybook is deployed, you can just pass the URL directly, no need for the complex recipe. That recipe is only for the case where you build your Storybook in CI and _don't_ deploy to a URL. I think the [README](https://github.com/storybookjs/test-runner/#running-in-ci) has a more accurate structure for these sections.

Reply:
Yea agreed, this section is misleading at the moment. Here's my initial draft: https://www.notion.so/chromatic-ui/Test-Runner-ddc1d47c3889491087d5cef1496f1c0c#1e3f97aed2e94d6fa654f8ac6ad9b182

The first option should be `1. Run against locally built Storybooks`. It's not testing a deployed storybook. It's building it on CI and testing that.

Reply:
With the latest commits. I've updated based on the conversation I and @kylegach had earlier today. @winkerVSbecks let me know if you're ok with it as well.